dev/README_RELEASE_PYTHON_CLIENT.md

```diff
@@ -0,0 +1,588 @@
+
+- [Release Process](#release-process)
+  - [Prepare PyPI convenience "RC" packages](#prepare-pypi-convenience-rc-packages)
+  - [Prepare Vote email on the Airflow Client release candidate](#prepare-vote-email-on-the-airflow-client-release-candidate)
+- [Verify the release candidate by PMC members](#verify-the-release-candidate-by-pmc-members)
+  - [SVN check](#svn-check)
+  - [Reproducible package check](#reproducible-package-check)
+  - [Signature check](#signature-check)
+  - [SHA512 checksum check](#sha512-checksum-check)
+- [Verify release candidates by Contributors](#verify-release-candidates-by-contributors)
+  - [Testing with Breeze's start-airflow](#testing-with-breezes-start-airflow)
+- [Publish the final Apache Airflow client release](#publish-the-final-apache-airflow-client-release)
+  - [Summarize the voting for the Apache Airflow client release](#summarize-the-voting-for-the-apache-airflow-client-release)
+  - [Publish release to SVN](#publish-release-to-svn)
+  - [Prepare PyPI "release" packages](#prepare-pypi-release-packages)
+  - [Create release on GitHub](#create-release-on-github)
+  - [Notify developers of release](#notify-developers-of-release)
+  - [Add release data to Apache Committee Report Helper](#add-release-data-to-apache-committee-report-helper)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+# Release Process
+
+The client versioning is independent of the Airflow versioning.
+
+The Python client is generated using Airflow's [openapi spec](https://github.com/apache/airflow/blob/master/clients/gen/python.sh).
+To update the client for new APIs do the following steps:
+
+```bash
+- Checkout the v2-*-test branch of Airflow where you generate the client from
+
+```bash
+# If you have not done so yet
+git clone git@github.com/apache/airflow
+cd airflow
+# Checkout the right branch
+git checkout v2-8-test
+export AIRFLOW_REPO_ROOT=$(pwd -P)
+cd ..
+```
+
+- Checkout the right branch (usually main) of the Airflow Python client where you
+  generate the source code to
+
+```bash
+# If you have not done so yet
+git clone git@github.com/apache/airflow-client-python
+cd airflow-client-python
+# Checkout the right branch
+git checkout main
+export CLIENT_REPO_ROOT=$(pwd -P)
+cd ..
+```
+
+- Set your version in `clients/python/version.txt` in the **Airflow** repository.
+  Set the `VERSION` and `VERSION_SUFFIX` environment variables. Note that version.txt should contain
+  the target version - without the suffix.
+
+```bash
+cd ${AIRFLOW_REPO_ROOT}
+VERSION="2.8.0"
+VERSION_SUFFIX="rc1"
+echo "${VERSION}" > clients/python/version.txt
+```
+
+- Get a diff between the last airflow version and the current airflow version
+  which this release is based on:
+
+```shell script
+cd ${AIRFLOW_REPO_ROOT}
+git log 2.3.0..HEAD --pretty=oneline -- airflow/api_connexion/openapi/v1.yaml
+```
+
+- Update CHANGELOG.md with the details.
+
+- Create PR and merge it to the `v2-*-stable` branch.
```

Comment:
We create this PR against Airflow repo or Airflow client repo? If Airflow, what about v2-*-test branch?

Reply:
That's one of the big changes here. You will now release from the Airflow repo (because this is where the open API specification is about - the Python client will be merely the repository of generated source code - for anyone to look at (they will also be able to build their client from there as well, but the whole "generale + package" right now will be done in the Airflow repo (no generated code of the client will be kept in the repo though).

Reply:
> If Airflow, what about v2-*-test branch?

I am going to cherry-pick that one to v2-test branch. When we release a new API client - it makes sense to release it using v2-stable version of the open api specification - because otherwise the v1.yaml might contain some endpoints that are only available in 2.9.

That's why it makes even more sense to release it from Airflow repository - because if we (for example) find any fix that needs to be applied to v1.yaml specification in main  - we just cherry-pick that change to v2-8-test, merge it to stable and relase from there (even if we decide to **just** release an API fix - without releasing Airflow from the stable branch.

Reply:
But let me clarify the process - because I did not inded mention that it first should be merged to main and then cherry-picked to v2-stable

Reply:
All right - updated/clarified - I am going to cherry-pick it and assist you with the process of release in case some fixes/adjustments will be needed.